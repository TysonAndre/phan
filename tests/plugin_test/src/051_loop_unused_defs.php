<?php

function foo51(int $value): bool {
    do {
        $a = true; // PhanUnusedVariable should not be emitted

        if ($value == 1) {
            break;
        }

        $a = false;
    } while (false);

    return $a;
}

function foo51b(): bool {
    foreach (['a', 'b'] as $x) {
        $a = true; // PhanUnusedVariable should not be emitted

        if ($x === 'b') {
            break;
        }

        $a = false;
    }

    return $a;
}


function foo51c(int $value) {
    $b = true; // PhanUnusedVariable should be emitted

    do {
        if ($value == 1) {
            $b = 'a string';
            echo $b;
            break;
        }
    } while (false);
}
foo51(1);
foo51b();
foo51c(1);
