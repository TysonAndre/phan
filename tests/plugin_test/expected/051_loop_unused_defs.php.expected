src/051_loop_unused_defs.php:33 PhanUnusedVariable Unused definition of variable $b
src/051_loop_unused_defs.php:56 PhanPluginUseReturnValueKnown Expected to use the return value of the user-defined function/method \foo51() defined at src/051_loop_unused_defs.php:3
src/051_loop_unused_defs.php:57 PhanPluginUseReturnValueKnown Expected to use the return value of the user-defined function/method \foo51b() defined at src/051_loop_unused_defs.php:17
